scripts/lib/sharding.py

```diff
@@ -19,15 +21,43 @@ def write_realm_nginx_config_line(f: Any, host: str, port: str) -> None:
     set $tornado_server http://tornado{port};
 }}\n""")
 
+def hash_sharding_config() -> str:
+    config_file = get_config_file()
+    if not config_file.has_section("tornado_sharding"):
+        return hashlib.sha256(b'').hexdigest()
+    contents = subprocess.check_output([
+        "crudini", "--get", "--format=lines", "/etc/zulip/zulip.conf", "tornado_sharding",
+    ])
+    return hashlib.sha256(contents).hexdigest()
+
 # Basic system to do Tornado sharding.  Writes two output .tmp files that need
 # to be renamed to the following files to finalize the changes:
 # * /etc/zulip/nginx_sharding.conf; nginx needs to be reloaded after changing.
 # * /etc/zulip/sharding.json; supervisor Django process needs to be reloaded
 # after changing.  TODO: We can probably make this live-reload by statting the file.
 #
 # TODO: Restructure this to automatically generate a sharding layout.
+
+parser = argparse.ArgumentParser(description="Adjust Tornado sharding configuration")
+parser.add_argument("--verify", action='store_true')
```

Comment:
Maybe we want a description here explaining that verify will return status 0 if the configuration is already correct?

Reply:
Absolutely; added.